compiler/parser.mly

```diff
@@ -138,16 +138,16 @@ stmt_list:
 
 stmt:
     expr ENDLINE                                { Expr($1) } 
-  | FOR LPAREN expr COMMA bool_expr COMMA
-    assignment_stmt RPAREN
-    LCURLY stmt_list RCURLY ENDLINE             { For($3, $5, $7, $10) } 
-  | WHILE LPAREN bool_expr RPAREN
-    LCURLY stmt_list RCURLY ENDLINE             { While($3, $6) }
+  | FOR LPAREN assignment_stmt COMMA bool_expr COMMA
```

Comment:
i think the reason we had `expr` before is in case your counter has been declared before. have we decided not to support that anymore?